clustermesh/clustermesh.go

```diff
@@ -50,7 +50,10 @@ import (
 const (
 	configNameClusterID   = "cluster-id"
 	configNameClusterName = "cluster-name"
-	configNameTunnel      = "tunnel"
+
+	configNameTunnelLegacy   = "tunnel"
```

Comment:
nit: Wondering if we should use `Deprecated` vs Legacy to indicate this feature will be axed in subsequent releases. Might make the intent clearer for devs and to grep when the time comes??

Reply:
The thing is that the feature is deprecated in Cilium, not Cilium CLI. Cilium CLI will have to support it even once it has been removed in Cilium, because Cilium CLI supports more than just one Cilium version. Thus my choice of using `Legacy`, which I think therefore fits better than `Deprecated`.

You however raise a good point - we probably should add a few comments saying that we can remove the logic once Cilium v1.14 is EOL (which will happen in about 2 years). Does that sound good to you?